### PR TITLE
Add finish transaction method

### PR DESCRIPTION
**Pull request by omninando**
### Added

- Finish transaction method with new gateway callback endpoint which receives an orderGroupId

#### How should this be manually tested?
`yarn test`

#### Screenshots or example usage
n/a

#### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: docs/checkout/README.en.md [hunk 1]
```diff
@@ -64,12 +61,12 @@ You can find out what all the sections are by taking a look at the `_allOrderFor
...
 
 ### **Example**
+
 ```html
-$(window).on('orderFormUpdated.vtex', function(evt, orderForm) {
-  alert('Someone changed the orderForm!');
-  console.log(orderForm);
-});
+$(window).on('orderFormUpdated.vtex', function(evt, orderForm) { alert('Someone
+changed the orderForm!'); console.log(orderForm); });
```
Comment (darlenedms) [inline docs/checkout/README.en.md]: Weird indentation, is that correct?

File: docs/checkout/README.en.md [hunk 2]
```diff
@@ -81,12 +78,12 @@ This is one of the most important methods: it is essential to make sure that an
...
 
 ### **Example**
+
 ```html
-vtexjs.checkout.getOrderForm()
-  .done(function(orderForm) {
-    console.log(orderForm);
-  });
+vtexjs.checkout.getOrderForm() .done(function(orderForm) {
+console.log(orderForm); });
```
Comment (darlenedms) [inline docs/checkout/README.en.md]: Ditto

File: docs/checkout/README.en.md [hunk 3]
```diff
@@ -112,51 +109,46 @@ Do not forget to use getOrderForm first.
...
-  }).done(function(orderForm) {
-    alert("Name changed!");
-    console.log(orderForm);
-    console.log(orderForm.clientProfileData);
-  })
+vtexjs.checkout.getOrderForm() .then(function(orderForm) { var clientProfileData
+= orderForm.clientProfileData; clientProfileData.firstName = William; return
+vtexjs.checkout.sendAttachment('clientProfileData', clientProfileData)
+}).done(function(orderForm) { alert("Name changed!"); console.log(orderForm);
+console.log(orderForm.clientProfileData); })
```
Comment (darlenedms) [inline docs/checkout/README.en.md]: Ditto

File: docs/checkout/README.en.md [hunk 4]
```diff
@@ -112,51 +109,46 @@ Do not forget to use getOrderForm first.
...
-  .then(function(orderForm) {
-    var obs = 'No onion!'
-    return vtexjs.checkout.sendAttachment('openTextField', { value: obs });
-  }).done(function(orderForm) {
-    console.log("openTextField filled with: ", orderForm.openTextField);
-  });
+vtexjs.checkout.getOrderForm() .then(function(orderForm) { var obs = 'No onion!'
+return vtexjs.checkout.sendAttachment('openTextField', { value: obs });
+}).done(function(orderForm) { console.log("openTextField filled with: ",
+orderForm.openTextField); });
```
Comment (darlenedms) [inline docs/checkout/README.en.md]: Ditto

File: docs/checkout/README.en.md [hunk 5]
```diff
@@ -175,32 +167,26 @@ Items that are already in the orderForm will remain unchanged.
...
-  seller: '1'
-};
-vtexjs.checkout.addToCart([item], null, 3)
-  .done(function(orderForm) {
-    alert('Item added!');
-    console.log(orderForm);
-  });
+var item = { id: 2000017893, quantity: 1, seller: '1' };
+vtexjs.checkout.addToCart([item], null, 3) .done(function(orderForm) {
+alert('Item added!'); console.log(orderForm); });
```
Comment (darlenedms) [inline docs/checkout/README.en.md]: Ditto

File: docs/checkout/README.en.md [hunk 6]
```diff
@@ -226,38 +212,28 @@ Do not forget to use getOrderForm first.
...
-    alert('Items updated!');
-    console.log(orderForm);
-  });
-  ```
-  
+vtexjs.checkout.getOrderForm() .then(function(orderForm) { var itemIndex = 0;
+var item = orderForm.items[itemIndex]; var updateItem = { index: itemIndex,
+quantity: 5 }; return vtexjs.checkout.updateItems([updateItem], null, false); })
+.done(function(orderForm) { alert('Items updated!'); console.log(orderForm); });
+```
```
Comment (darlenedms) [inline docs/checkout/README.en.md]: Ditto

File: docs/checkout/README.md [hunk 7]
```diff
@@ -1182,4 +1182,28 @@ vtexjs.checkout.replaceSKU(items)
   .then(function(orderForm) {
     console.log("Novos items: ", orderForm.items);
   });
+```
+
+## finishTransaction(orderGroupId, expectedOrderFormSections)
+
+Avisa a API do checkout para terminar uma transação e ir para a url de final (e.g. `order-placed`, `checkout`).
```
Comment (klzns) [inline docs/checkout/README.md]:
```suggestion
Avisa a API do checkout para terminar uma transação e ir para a url final (e.g. `order-placed`, `checkout`).
```

Comment (omninando) [reply docs/checkout/README.md]: good catch!

**Pull request merged**